server.js

```diff
@@ -1,30 +1,74 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
-//   PORT=9000 npm start
+// PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+const listEndpoints = require('express-list-endpoints')
+
 // Start defining your routes here
+
+// Root endpoint
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Route for array of all books with pagination as default
+app.get('/books', (req, res) => {
+  const { page, sort, author, title } = req.query
+  let booksList = booksData
+
+  // sort books by rating using query, returns sorted booksList
+  if (sort === 'rating_dsc') {
+    booksList = booksList.sort((a, b) => (b.average_rating - a.average_rating))
+  } else if (sort === 'rating_asc') {
+    booksList = booksList.sort((a, b) => (a.average_rating - b.average_rating))
+  }
```

Comment:
Because you have a value here called `rating_asc`, I feel like there might be a `title_desc` in the future.  Was that the idea with the naming?

Reply:
Had an idea of doing more, ended up with just this one though. Added some more now!